modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -6920,7 +6928,7 @@ private void _validateObjectStateTransition(
 
 		Serializable existingValue = existingValues.get(objectField.getName());
 
-		if (existingValue == null) {
+		if ((existingValue == null) || existingValue.equals(StringPool.BLANK)) {
```

Comment:
When comparing Strings, we usually use this `Objects.equals()`

`Objects.equals(existingValue, StringPool.BLANK)`